easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,231 @@
+        res = version_regex.search(txt)
+        if res:
+            setting = res.group(1)
+            self.log.debug("Extracted OpenMPI setting %s: '%s' from search text", search, setting)
+            return setting
+        else:
+            raise EasyBuildError("Failed to extract OpenMPI setting '%s' using regex pattern '%s' from: %s",
+                                 search, version_regex.pattern, txt)
+
+    def __init__(self, *args, **kwargs):
+        """Extra initialization: determine system MPI version, prefix and any associated envvars."""
+        super(SystemMPI, self).__init__(*args, **kwargs)
+
+        # Determine MPI path (real path, with resolved symlinks) to ensure it exists
+        mpi_name = self.cfg['name'].lower()
+
+        # Check we have MPI wrappers
+        if mpi_name != 'impi':
+            mpi_c_wrapper = 'mpicc'
+        else:
+            mpi_c_wrapper = 'mpiicc'
+        path_to_mpi_c_wrapper = which(mpi_c_wrapper)
+        if path_to_mpi_c_wrapper:
+            path_to_mpi_c_wrapper = os.path.realpath(path_to_mpi_c_wrapper)
+            self.log.info("Found path to MPI implementation '%s' %s compiler (with symlinks resolved): %s",
+                          mpi_name, mpi_c_wrapper, path_to_mpi_c_wrapper)
+        else:
+            raise EasyBuildError("%s not found in $PATH", mpi_c_wrapper)
+
+        # Determine compiler version and installation prefix
+        if mpi_name == 'openmpi':
+            output_of_ompi_info, _ = run_cmd("ompi_info", simple=False)
+            # Extract the version of OpenMPI
+            self.mpi_version = self.extract_ompi_setting("Open MPI", output_of_ompi_info)
+
+            # Extract the installation prefix
+            self.mpi_prefix = self.extract_ompi_setting("Prefix", output_of_ompi_info)
+
+            # Extract any OpenMPI environment variables in the current environment and ensure they are added to the
+            # final module
+            raw_env = os.environ
+            self.mpi_envvars = dict((key, value) for key, value in raw_env.iteritems() if key.startswith("OMPI_"))
+
+            # Extract the C compiler used underneath OpenMPI, check for the definition of OMPI_MPICC
+            self.mpi_c_compiler = self.extract_ompi_setting("C compiler", output_of_ompi_info)
+
+        elif mpi_name == 'impi':
+            # Extract the version of IntelMPI
+            # The prefix in the the mpiicc script can be used to extract the explicit version
+            contents_of_mpiicc, _ = read_file(path_to_mpi_c_wrapper)
+            prefix_regex = re.compile(r'(?<=compilers_and_libraries_)(.*)(?=/linux/mpi)', re.M)
+            self.mpi_version = prefix_regex.search(contents_of_mpiicc)
+
+            # Extract the installation prefix
+            # If I_MPI_ROOT is defined, let's use that
+            raw_env = os.environ
+            if raw_env['I_MPI_ROOT']:
+                self.mpi_prefix = raw_env['I_MPI_ROOT']
```

Comment:
no need to introduce `raw_env`, and you can do this in one go:

```python
self.mpi_prefix = os.getenv('I_MPI_ROOT', os.path.dirname(os.path.dirname(mpi_c_wrapper)))
```

In any case: also check that the directory you come up with actually exists?